### PR TITLE
recent-blocks - dont listen for block when on infura providers

### PR DESCRIPTION
**Pull request by frankiebee**
this will remove the listener off the block tracker when on infura providers sense it is not needed 

that will allow the block tracker to stop polling

### CODE REVIEW

File: app/scripts/controllers/recent-blocks.js [hunk 1]
```diff
@@ -33,13 +41,30 @@ class RecentBlocksController {
...
-
-    this.blockTracker.on('latest', async (newBlockNumberHex) => {
+    const blockListner = async (newBlockNumberHex) => {
       try {
         await this.processBlock(newBlockNumberHex)
       } catch (err) {
         log.error(err)
       }
+    }
+    let isListeng = false
```
Comment (whymarrh) [inline app/scripts/controllers/recent-blocks.js]: Small typo, should be `isListening`

**Review by whymarrh — DISMISSED** [review]

File: app/scripts/controllers/recent-blocks.js [hunk 2]
```diff
@@ -33,13 +41,30 @@ class RecentBlocksController {
...
+    const { type } = networkController.getProviderConfig()
+    if (!INFURA_PROVIDER_TYPES.includes(type) && type !== 'loading') {
+      this.blockTracker.on('latest', blockListner)
+      isListeng = true
```
Comment (tmashuang) [inline app/scripts/controllers/recent-blocks.js]:
```suggestion
      isListening = true
```

**Review by tmashuang — CHANGES_REQUESTED** [review]

File: app/scripts/controllers/recent-blocks.js [hunk 3]
```diff
@@ -33,13 +41,30 @@ class RecentBlocksController {
...
+    }
+    networkController.on('networkDidChange', (newType) => {
+      if (INFURA_PROVIDER_TYPES.includes(newType) && isListeng) {
```
Comment (tmashuang) [inline app/scripts/controllers/recent-blocks.js]:
```suggestion
      if (INFURA_PROVIDER_TYPES.includes(newType) && isListening) {
```

File: app/scripts/controllers/recent-blocks.js (hunk 1 above)
Comment (tmashuang) [inline app/scripts/controllers/recent-blocks.js]:
```suggestion
    let isListening = false
```

File: app/scripts/controllers/recent-blocks.js [hunk 4]
```diff
@@ -33,13 +41,30 @@ class RecentBlocksController {
...
+        this.blockTracker.removeListener('latest', blockListner)
+      } else if (
+        !INFURA_PROVIDER_TYPES.includes(type) &&
+        type !== 'loading' &&
+        !isListeng
```
Comment (tmashuang) [inline app/scripts/controllers/recent-blocks.js]:
```suggestion
        !isListening
```

**Review by tmashuang — CHANGES_REQUESTED** [review]

**Review by tmashuang — APPROVED** [review]

**Pull request merged**